src/NeoSharp.Core/DI/Modules/NetworkModule.cs

```diff
@@ -36,9 +35,22 @@ public void Register(IContainerBuilder containerBuilder)
                             t != typeof(MessageHandlerProxy))
                 .ToArray();
 
-            containerBuilder.Register(typeof(IMessageHandler<>), messageHandlerTypes);
-            containerBuilder.RegisterInstanceCreator<IMessageHandler<Message>>(c =>
-                new MessageHandlerProxy(c, messageHandlerTypes, c.Resolve<ILogger<MessageHandlerProxy>>()));
+            containerBuilder.RegisterCollection(typeof(IMessageHandler<>), messageHandlerTypes);
+            containerBuilder.RegisterCollection(typeof(IMessageHandler), new[]
+            {
+                typeof(VersionMessageHandler),
```

Comment:
Now we need to remember to add it here. In previous version all handlers were discoverable based on message type.

Reply:
That can be done too ... and add to this list